x-pack/plugins/infra/public/pages/logs/page_template.tsx

```diff
@@ -17,8 +23,35 @@ export const LogsPageTemplate: React.FC<LazyObservabilityPageTemplateProps> = (
       observability: {
         navigation: { PageTemplate },
       },
+      docLinks,
     },
   } = useKibanaContextForPlugin();
 
-  return <PageTemplate {...pageTemplateProps} />;
+  const { http } = useKibana().services;
+  const basePath = http!.basePath.get();
+
+  const { sourceStatus } = useLogSourceContext();
+  const noDataConfig: KibanaPageTemplateProps['noDataConfig'] =
+    sourceStatus?.logIndexStatus !== 'missing'
```

Comment:
We had some a chat about this approach within the team and we think it's better to move this condition out of the  `<LogsPageTemplate />` component and transform it into a prop. The consumers then will decide if they want to show a `noData` version of the template.

```tsx
interface LogsPageTemplateProps extends LazyObservabilityPageTemplateProps {
  hasData: boolean;
}

// In case the consumer forgets to pass the prop or just wants to use the wrapping template we default to `true`
export const LogsPageTemplate: React.FC<LogsPageTemplateProps> = ({ hasData = true, ... }) => {
  // ...

  const noDataConfig = hasData ? undefined : {...}
  // ...
} 
```

This would solve a problem where the settings page cannot be accessed if there's no data.

Reply:
/cc @Zacqary